lib/cartopy/img_transform.py

```diff
@@ -222,6 +222,9 @@ def _determine_bounds(x_coords, y_coords, source_cs):
     return bounds
 
 
+KDTREE_CACHE = {}
```

Comment:
I've also added a very basic cache for the KDTree used in `regrid`, because the image redraw doesn't work properly otherwise.

Reply:
Yup, it should; there's no way to properly define caching the kdtree here. But the cache was useful for the PoC, so made it into the changes proposed here.

The redraw simply doesn't work unless the kdtree is cached. The wrong part of the image is drawn or the whole original image is mis-scaled if the original build of the kdtree isn't cached. The original state of the kdtree is important to all image redraws, regardless of intermediate redraws, which I guess makes sense.